### PR TITLE
Fix parsing of filesystem URIs on Windows

### PR DESCRIPTION
**Pull request by JanTvrdik**
Converts `file://C:\json-schema\draft-03\schema` to valid `file:///C:/json-schema/draft-03/schema`

### CODE REVIEW

**Comment by stefk** [comment]
I'm not sure it's a good idea to make such a conversion inside the validator. All JSON Schema specs require URIs to adhere to RFC3986, so an error is actually expected in your example.

**Comment by JanTvrdik** [comment]
The alternative solution is to fix all tests to construct to correct URLs on Windows which is a lot more work, because suddenly you cannot do `'file://' . $diskPath`.

**Comment by stefk** [comment]
Ah ok, so the problem is actually that the test suite is broken on Windows due to those concatenations? If that's the case, I'd really prefer fixing the issue at this level. Your code could be reused to create a helper method for that purpose.

I'll try to have a look at it this weekend.

**Comment by JanTvrdik** [comment]
This is an alternative implementation, unfortunately it still requires modification of Uri class to fix missing slash in path after processing with parse_url.

File: src/Uri.php [hunk 1]
```diff
@@ -200,6 +200,10 @@ private function buildFromRawUri($rawUri)
             }
         }
 
+        if ($this->parts['scheme'] === 'file' && preg_match('/^[A-Z]:/', $this->parts['path'])) {
+            $this->parts['path'] = '/' . strtr($this->parts['path'], '\\', '/');
```
Comment (stefk) [inline src/Uri.php]: I understand the need for the leading slash, but I don't think there should be a `strtr` at this point.

**Comment by stefk** [comment]
Thanks!

**Pull request merged**